File: boj/boj1325.py
import sys
import collections
input = sys.stdin.readline

V, E = list(map(int, input().split()))
graph = [ [] for _ in range( V + 1)]
for _ in range(E) :
    (end, start) = list(map(int, input().split()))
    graph[start].append(end)

res = [-1]

def bfs(root) :
    q = collections.deque([root])
    tempRes = 1
    visited = [False] * (V + 1)
    visited[root] = True
    while q :
        cur = q.popleft()
        for next in graph[cur] :
            if not visited[next] :
                visited[next] = True
                tempRes += 1
                q.append(next)
    return tempRes
for ind in range(1, V + 1) :
    res.append(bfs(ind))
maxVal = max(res)
sRes = []
for ind in range(V + 1) :
    if res[ind] == maxVal :
        sRes.append(ind)
print(*sRes)

File: boj/boj1325.swift
import Foundation

func sol() {
    let NM = readLine()!.split(separator: " ").map{Int($0)!}
    let (N, M) = (NM[0], NM[1])
    
    var graph: [[Int]] = Array(repeating: [] , count : N  + 1)
    for _ in 0..<M {
        let endAndStart = readLine()!.split(separator: " ").map{Int($0)!}
        graph[endAndStart[1]].append(endAndStart[0])
    }

    var visited = Array(repeating: false, count: N + 1)
    func dfs(_ root: Int) -> Int {
        
        var currentDepth = 1
        for next in graph[root] {
            if visited[next] == false {
                visited[next] = true
                currentDepth += dfs(next)
            }
        }
        return currentDepth
    }

    var res: [Int] = [-1]
    for ind in 1...N {
        visited = Array(repeating: false, count: N + 1)
        visited[ind] = true
        res.append(dfs(ind))
    }
    let maxVal = res.max()!
    print(res.enumerated().filter{$0.1 == maxVal}.map{ String($0.0) }.joined(separator: " "))
}
sol()